File: tailwind-admin-reactjs-free/package/src/lib/matricula.ts
import { api } from './api';
import type { EstudianteMatricula } from 'src/types/matricula';

export function listSinGrupo(): Promise<EstudianteMatricula[]> {
  return api.get<EstudianteMatricula[]>('/matricula/sin-grupo');
}

export function listPorGrupo(idGrupo: number): Promise<EstudianteMatricula[]> {
  return api.get<EstudianteMatricula[]>(`/matricula/grupo/${idGrupo}`);
}

export function setGrupoEstudiante(
  idEstudiante: number,
  idGrupo: number | null,
): Promise<EstudianteMatricula> {
  return api.put<EstudianteMatricula>(`/matricula/estudiante/${idEstudiante}`, {
    id_grupo: idGrupo,
  });
}

export function matriculaMasiva(
  idGrupo: number,
  estudiantesIds: number[],
): Promise<EstudianteMatricula[]> {
  return api.post<EstudianteMatricula[]>('/matricula/masiva', {
    id_grupo: idGrupo,
    estudiantes_ids: estudiantesIds,
  });
}

File: backend/app/schemas/matricula.py
from pydantic import BaseModel, Field


class EstudianteMatricula(BaseModel):
    id_estudiante: int
    name_estudiante: str
    sec_name_estudiante: str
    num_documento_estudiante: str
    id_grupo: int | None
    grupo: str | None


class MatriculaUpdate(BaseModel):
    id_grupo: int | None = None


class MatriculaMasiva(BaseModel):
    id_grupo: int
    estudiantes_ids: list[int] = Field(min_length=1)
